bucket.go

```diff
@@ -98,6 +98,9 @@ type BucketService interface {
 	FindBucketByName(ctx context.Context, orgID ID, name string) (*Bucket, error)
 }
 
+// BucketMiddleware is a service middleware func.
+type BucketMiddleware func(BucketService) BucketService
```

Comment:
Usually any middleware definition is concrete. The reason is because most middleware don't share a common signature.